configuration.rst

```diff
@@ -24,159 +24,334 @@ The configuration file must contain a set of key value pairs. At minimum you mus
+  **In-Database** `n/a`
+  =============== =================
+
    Time to live, in seconds, for an idle database pool connection. If the timeout is reached the connection will be closed.
    Once a new request arrives a new connection will be started.
 
-.. _db-extra-search-path:
+.. _db-pre-request:
 
-db-extra-search-path
---------------------
+db-pre-request
+--------------
 
-  Extra schemas to add to the `search_path <https://www.postgresql.org/docs/current/ddl-schemas.html#DDL-SCHEMAS-PATH>`_ of every request. These schemas tables, views and stored procedures **don't get API endpoints**, they can only be referred from the database objects inside your :ref:`db-schema`.
+  *For backwards compatibility, this config parameter is also available without prefix as "pre-request".*
 
-  This parameter was meant to make it easier to use **PostgreSQL extensions** (like PostGIS) that are outside of the :ref:`db-schema`.
+  =============== =================
+  **Environment** PGRST_DB_PRE_REQUEST
+  **In-Database** pgrst.db_pre_request
+  =============== =================
 
-  Multiple schemas can be added in a comma-separated string, e.g. ``public, extensions``.
+  A schema-qualified stored procedure name to call right after switching roles for a client request. This provides an opportunity to modify SQL variables or raise an exception to prevent the request from completing.
 
-.. _db-channel:
+.. _db-prepared-statements:
 
-db-channel
+db-prepared-statements
+----------------------
+
+  =============== =================
+  **Environment** PGRST_DB_PREPARED_STATEMENTS
+  **In-Database** pgrst.db_prepared_statements
+  =============== =================
+
+  Enables or disables prepared statements.
+
+  When disabled, the generated queries will be parameterized (invulnerable to SQL injection) but they will not be prepared (cached in the database session). Not using prepared statements will noticeably decrease performance, so it's recommended to always have this setting enabled.
+
+  You should only set this to ``false`` when using PostgresSQL behind an external connection pooler such as PgBouncer working in transaction pooling mode. See :ref:`this section <external_connection_poolers>` for more information.
+
+.. _db-root-spec:
+
+db-root-spec
+------------
+
+  *For backwards compatibility, this config parameter is also available without prefix as "root-spec".*
+
+  =============== =================
+  **Environment** PGRST_DB_ROOT_SPEC
+  **In-Database** pgrst.db_root_spec
+  =============== =================
+
+  A schema-qualified stored procedure to call instead of the built-in OpenAPI code on `/`. This provides the opportunity to implement a custom response on the root endpoint.
+
+.. _db-schemas:
+
+db-schemas
 ----------
 
-  The name of the notification channel that PostgREST uses for :ref:`schema_reloading` and configuration reloading.
+  *For backwards compatibility, this config parameter is also available in singular as "db-schema".*
```

Comment:
As noticed in https://github.com/PostgREST/postgrest-docs/issues/487#issuecomment-1007938516, in-database configuration does not support the old aliases. Maybe we should add a hint here.

Reply:
Went without a hint for now. I think having the new names used everywhere should be enough to make people use it.